oxrdflib/parser.py

```diff
@@ -0,0 +1,131 @@
+import warnings
+from typing import Any, Optional
+
+from rdflib import ConjunctiveGraph, Graph
+from rdflib.exceptions import ParserError
+from rdflib.parser import (
+    FileInputSource,
+    InputSource,
+    Parser,
+    URLInputSource,
+    create_input_source,
+)
+
+from oxrdflib._converter import ox_to_rdflib_type, rdflib_to_mime_type, to_ox
+from oxrdflib.store import OxigraphStore
+
+__all__ = [
+    "OxigraphTurtleParser",
+    "OxigraphNTriplesParser",
+    "OxigraphRdfXmlParser",
+]
+
+
+class OxigraphParser(Parser):
+    def __init__(self):
+        pass
+
+    def parse(
+        self,
+        source: InputSource,
+        sink: Graph,
+        format: str,
+        encoding: Optional[str] = "utf-8",
+        **kwargs: Any,
+    ) -> None:
+        if encoding not in (None, "utf-8"):
+            raise ParserError("N3/Turtle files are always utf-8 encoded, I was passed: %s" % encoding)
+
+        if not isinstance(sink.store, OxigraphStore):
+            warnings.warn(
+                "Graph store should be an instance of OxigraphStore, "
+                f"got {type(sink.store).__name__} store instead."
+                " Attempting to parse using rdflib native parser.",
+                stacklevel=2,
+            )
+            sink.parse(source, format=ox_to_rdflib_type(format))
+
+        else:
+            base_iri = sink.absolutize(source.getPublicId() or source.getSystemId() or "")
+
+            if isinstance(source, FileInputSource):
+                input = source.file
+            elif isinstance(source, URLInputSource):
+                input = create_input_source(source.url, format=ox_to_rdflib_type(format)).getByteStream()
```

Comment:
For my own understanding: Isn't this call returning you an other `URLInputSource`?
Is it needed to make sure the format is one understood by rdflib?

Reply:
Indeed, as otherwise it gets our `ox` formats , and then perform request accepting any format.

Reply:
In this was we are forcing correct `rdflib` format (thus mime type).

Reply:
Thank you!